amt.go

```diff
@@ -285,21 +285,25 @@ func (n *Node) forEachAt(ctx context.Context, bs cbor.IpldStore, height int, sta
 
 	subCount := nodesForHeight(height)
 	for i, v := range n.expLinks {
-		if v != cid.Undef {
-			offs := offset + (uint64(i) * subCount)
-			nextOffs := offs + subCount
-			if start >= nextOffs {
-				continue
-			}
-
-			var sub Node
+		var sub Node
```

Comment:
Do we really want to copy the node?

Reply:
Ok, I talked this over with @whyrusleeping and _not_ doing this could end up caching the entire tree in memory.

IMO, the _right_ way to do this would be to have some kind of "max cached nodes" limit, and flush when we hit that limit, but that's a larger refactor.